auacm/app/static/js/controllers/ScoreboardController.js

```diff
@@ -61,38 +13,95 @@ app.controller('ScoreboardController', ['$scope', '$http', '$routeParams', '$win
             $scope.competition = response.data.data.competition;
             $scope.compProblems = response.data.data.compProblems;
             $scope.teams = response.data.data.teams;
-            genScoreboard();
+            genScoreboard($scope);
+            connectToSocket($window, $scope);
 
             // TODO(brandonlmorris) Add offset to server/client times
-            $scope.timeLeft = $scope.competition.length;
-            $scope.competition.length = $scope.competition.length * 1000;
-            var thisTime = new Date().getTime();
-            var Timer = $interval(function() {
-                $scope.timeLeft = ($scope.competition.startTime + $scope.competition.length - thisTime) / 1000;
-                thisTime = new Date().getTime();
-                if ($scope.timeLeft <= 0) {
-                    $scope.timeLeft = 0;
-                    $interval.cancel(Timer);
-                }
-            }, 1000);
+            var thisTime = Math.floor(new Date().getTime() / 1000);
+            if (thisTime < $scope.competition.startTime + $scope.competition.length) {
+                // if the competition is still active
+                $scope.active = true;
+                $scope.timeLeft = $scope.competition.startTime + $scope.competition.length
+                        - thisTime;
+                var timer = $interval(function() {
+                    $scope.timeLeft = $scope.competition.startTime + $scope.competition.length 
+                            - thisTime;
+                    thisTime = Math.floor(new Date().getTime() / 1000);
+                    if ($scope.timeLeft <= 0) {
+                        $scope.active = false;
+                        $scope.timeLeft = 0;
+                        $interval.cancel(timer);
+                    }
+                }, 1000);
+            }
+            
         },
         function(error) {
             
         });
+}]);
 
-
-    // Open the socket connection
-    var socket = io.connect('http://' + $window.location.host + '/judge')
-    socket.on('connect', function() {
-        console.log('connected');
+var genScoreboard = function($scope) {
```

Comment:
I moved it outside of the controller, because I thought it kind of made it look cleaner, but I'm not sure if it's the best route to go. I forgot about scoping, so we don't necessarily want to make these functions available. I'll move them back.